app/models/manageiq/providers/embedded_terraform/automation_manager/configuration_script_source.rb

```diff
@@ -85,22 +87,51 @@ def find_templates_in_git_repo
               .group_by          { |file| File.dirname(file) }
               .select            { |_dir, files| files.any? { |f| f.end_with?(".tf", ".tf.json") } }
               .transform_values! { |files| files.map { |f| File.basename(f) } }
-              .each do |parent_dir, files|
-        name = self.class.template_name_from_git_repo_url(git_repository.url, parent_dir)
+              .each do |relative_path, files|
+        name = self.class.template_name_from_git_repo_url(git_repository.url, relative_path)
+
+        template_full_path = File.join(git_checkout_tempdir, relative_path)
 
-        # TODO: add parsing for input/output vars
-        input_vars  = nil
-        output_vars = nil
+        input_vars, output_vars, terraform_version = parse_vars_in_template(template_full_path)
 
         template_dirs[name] = {
-          :relative_path => parent_dir,
-          :files         => files,
-          :input_vars    => input_vars,
-          :output_vars   => output_vars
+          :relative_path     => relative_path,
+          :files             => files,
+          :input_vars        => input_vars,
+          :output_vars       => output_vars,
+          :terraform_version => terraform_version,
         }
       end
     end
 
     template_dirs
+  rescue => error
+    _log.error("Failing scaning for terraform templates in the git repo: #{error}")
+    raise error
```

Comment:
Minor, but you don't need to `raise error` to re-raise the same exception in a rescue you can simply `raise` and it'll reraise the same error (won't hold up this PR for this though)